docs/website/contents/for-developers/Whitepaper.md

```diff
@@ -0,0 +1,244 @@
+    A("Consensus") --> C("Select From Candidate Chains")
+    A("Consensus") --> D("Accept transactions into Mempool")
+    A("Consensus") --> E("Store and Serve Historical Chain")
+
+
+    B("Mint Blocks") --> B1("Ask Ledger to Validate Transations")
+
+    C("Select From Candidate Chains") --> A1("Fetch Chains from Peers")
+```
+
+The Consensus layer is responsible to choose among the different chains that might co-exist on the network. The candidate chains may arise both from honest and adversarial participation. The Consensus Layer must implement the consensus protocol or the current Cardano era to know what is the right candidate chain to choose. To replicate the blockchain across all Cardano nodes, the Consensus layer must distribute the selected chain to its peers.
+
+The Consensus layer is also responsible for producing new blocks according to the current consensus protocol. For the new blocks to include meaningful data, Consensus has to receive and maintain a list of submitted transactions that are not in the chain yet.
+
+The following table outlines the informal components of the Haskell reference implementation and their responsibilities:
+
+| Component                                      | Responsibility                                                                      |
+|:-----------------------------------------------|:------------------------------------------------------------------------------------|
+| Forging Loop (aka The Mint)                    | Extend the Cardano chain by minting new blocks                                      |
+| `Mempool`                                      | Maintain a list of valid transactions to include in new blocks                      |
+| `ChainSync` and `BlockFetch` clients           | Gather candidate chains from peers                                                  |
+| `ChainSync` and `BlockFetch` servers           | Provide peers with access to `ChainDB` (see Storage Layer)                          |
+| `ChaineSel`                                    | Among all forks that can be fetched from peers, must identify the best chain.       |
+| N2N `TxSubmission`, N2C `TxSubmission` servers | Accept transactions from peers and store them into `Mempool` for further processing |
+|                                                |                                                                                     |
+|                                                |                                                                                     |
+|                                                |                                                                                     |
+
+## Responsibilities of the Storage Layer
+
+The Consensus layer needs to keep track of a significant amount of data to maintain the Cardano network. Some of this data is "hot" and relates to the candidate chains and other data is cold and relates to the historical chain.
```

Comment:
If you make this distinction, I would clarify what data is Hot in the diagram and what is cold